### DISCUSSION TITLE
Flashing Gap8 Bootloader

### DISCUSSION THREAD

**Discussion by Senpia368** [bitcraze/discussions#823] [Q&A] [answered]
Hello
I can't figure out which programmer I should use to flash the gap8 bootloader to the AI-deck and what adapters I need to plug it to my laptop. Can someone provide me with the links of what I should get and walk me through the steps?

**Comment by ataffanel** [comment, marked as answer]
Gap8 can be flashed with a J-link or a  ARM-USB-TINY-H JTAG debug probe. Flashing the bootloader is documented there: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader and the connection of the debug probe to the deck is documented there: https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/flashing/#jtag-link-cable-connection. The deck uses the ARM's [Coresight10](https://developer.arm.com/documentation/dui0499/d/ARM-DSTREAM-Target-Interface-Connections/CoreSight-10) connector pinout which is now quite standard (our debug adapter kit has it and at least olimex sells an adapter for it).

**Reply by Senpia368** [reply]
Thank you. I appreciate your help.
